coreUtilities/src/main/java/org/opensearch/migrations/transformation/entity/Entity.java

```diff
@@ -0,0 +1,19 @@
+package org.opensearch.migrations.transformation.entity;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Common interface for entities that are used to perform transformations
+ *
+ * Notes on performance: ObjectNode requires a fully marshalled json, for
+ * less than 1mb json bodies this will likely be fine; however, if dealing with larger
+ * objects or perf issues are encountered should consider adding an alternative
+ * path for JsonParser.  This might be needed for processing document bodies
+ * specifically.
+ */
+public interface Entity {
```

Comment:
Entity is _really_ generic.  I know that you're namespaced, but how likely will this bleed into other areas (like Http, or anything else)?